iothub/service/src/IotHubConnection.cs

```diff
@@ -296,6 +293,12 @@ private async Task<AmqpSession> CreateSessionAsync(TimeSpan timeout)
             }
             finally
             {
+                if (transport != null
+                    && transport is ClientWebSocketTransport clientWebSocketTransport)
+                {
+                    clientWebSocketTransport.Dispose();
```

Comment:
@abhipsaMisra is this safe to do? I'm not entirely sure what this method is doing, but it doesn't seem to save off any of the things it has created here, so it feels like they should only be kept for the duration of the method.

Reply:
I don't think we should dispose it here, we are giving a reference of this web socket transport to the amqp library, which it will hold on to until the connection is active. I think we should dispose the transport object when we dispose the iot hub connection

Reply:
Let's do a call later today, because I'm not seeing where the objects are being saved off.